src/main/java/org/openstreetmap/atlas/checks/validation/intersections/EdgeCrossingEdgeCheck.java

```diff
@@ -105,22 +109,46 @@ public boolean validCheckForObject(final AtlasObject object)
         // edges from being flagged more than once.
         final Set<Edge> collectedEdges = new EdgeCrossingEdgeWalker((Edge) object,
                 this.getInvalidCrossingEdges()).collectEdges();
-
         if (collectedEdges.size() > 1)
         {
-            final CheckFlag newFlag = new CheckFlag(getTaskIdentifier(object));
-            this.markAsFlagged(object.getIdentifier());
-            newFlag.addObject(object);
-            newFlag.addInstruction(this.getLocalizedInstruction(0, object.getOsmIdentifier()));
-
-            collectedEdges.forEach(invalidEdge ->
+            final List<Tuple2<Long, Set<Edge>>> edgeCrossPairs = collectedEdges
+                    .stream().filter(
+                            e -> e.getOsmIdentifier() != object.getOsmIdentifier())
+                    .map(e -> new Tuple2<>(e.getOsmIdentifier(),
+                            new EdgeCrossingEdgeWalker(e, this.getInvalidCrossingEdges())
+                                    .collectEdges()))
+                    .collect(Collectors.toList());
+            edgeCrossPairs.add(new Tuple2<>(object.getOsmIdentifier(), collectedEdges));
+            final Optional<Tuple2<Long, Set<Edge>>> maxPair = edgeCrossPairs.stream().max(
+                    (entry1, entry2) -> Integer.compare(entry1._2().size(), entry2._2().size()));
+            if (maxPair.isPresent())
             {
-                this.markAsFlagged(invalidEdge.getIdentifier());
-                newFlag.addObject(invalidEdge);
-                newFlag.addInstruction(
-                        this.getLocalizedInstruction(1, invalidEdge.getOsmIdentifier()));
```

Comment:
I think we still want to do this for each crossing Edge. This will help cut down on duplicate processing and give clearer instructions on what needs to be changed.

Reply:
As we discussed added intersecting edges locations to the flag.